prov/shm/src/smr_rma.c

```diff
@@ -157,8 +157,30 @@ ssize_t smr_generic_rma(struct smr_ep *ep, const struct iovec *iov,
 	total_len = ofi_total_iov_len(iov, iov_count);
 
 	smr_generic_format(cmd, peer_id, op, 0, data, op_flags);
-	if (total_len > SMR_INJECT_SIZE || (!smr_env.disable_cma &&
-	    (op != ofi_op_write || op_flags & FI_DELIVERY_COMPLETE))) {
+	if (total_len <= SMR_MSG_DATA_LEN && op == ofi_op_write &&
+	    !(op_flags & FI_DELIVERY_COMPLETE)) {
+		smr_format_inline(cmd, iov, iov_count);
+	} else if (total_len <= SMR_INJECT_SIZE &&
+		   (smr_env.disable_cma || op == ofi_op_write) &&
+		   !(op_flags & FI_DELIVERY_COMPLETE)) {
```

Comment:
The check for CMA and op were put there because of the lack of fallback protocols. 
Now it can just be:
```
else if (total_len <= SMR_INJECT_SIZE && !(op_flags & FI_DELIVERY))
```

Reply:
Hmm. I thought we still want to use iov protocol for read if CMA is available.
With your suggested change, do you mean we prefer inject protocol instead of iov protocol for medium/small RMA read even though CMA is available?

Reply:
Updated with your suggested change. Now
- inline  -  small size write
- inject  -  small size read, medium size read/write
- iov      -   large size read/write (w. CMA)
- mmap -  large size read/write (w/o CMA)